magda-web-client/src/helpers/CsvDataLoader.ts

```diff
@@ -71,6 +71,16 @@ class CsvDataLoader {
      */
     private toBeAbort: boolean = false;
 
+    /**
+     * We set this property to skip the next `complete` call.
+     * This would be usuful when we want to retry the data downloading.
```

Comment:
I still don't quite get why we skip the next complete call - is it because otherwise `complete` will get called with the wrong data?

Reply:
What happened was:
- We determined that, somehow, PapaParser's auto line ending detection doesn't work for the file (Around line 182) and planned to re-process the file with fixed line-ending setting "\n".
- We firstly, called `parser.abort();` and then call `resolve(retryLater(...))` to restart the whole process
  - Unfortunately, `parser.abort();` will make PapaParser trigger `onComplete` event and call my `complete` method as well (I thought it won't previously)
  -  in the `complete` method, the `resolve(result)` is called and make the promise be resolved earlier while the second retry still processes the file.